client/editSchedule.js

```diff
@@ -190,46 +158,31 @@ Template.addEditSomethingModal.helpers({
 });
 
 AutoForm.hooks({
-  editRoundForm: {
+  editEventForm: {
     onSubmit: function(insertDoc, updateDoc, currentDoc) {
       this.event.preventDefault();
 
       if(currentDoc._id) {
-        // Updating an existing round
-        Rounds.update({
-          _id: currentDoc._id,
-        }, {
-          $set: {
-            title: insertDoc.title,
-            startMinutes: insertDoc.startMinutes,
-            durationMinutes: insertDoc.durationMinutes,
-          }
-        });
-        $('#addEditSomethingModal').modal('hide');
+        delete updateDoc.$set.competitionId; // Don't touch the competitionId field. We don't trust browser input that much.
```

Comment:
The `updateDoc` parameter is conveniently made by the framework to pass on to Collection.update(). But it contains the `competitionId`, which we don't allow to be changed on the server.

So removing that makes the update 'legal', and retains most of the convenience (and maintainability).

It's a bit like throwing away your forbidden items before going through TSA.